great_expectations/experimental/datasources/interfaces.py

```diff
@@ -339,7 +344,13 @@ class Datasource(
         "type",
         "execution_engine",
         "assets",
-        "base_directory",
+        "base_directory",  # filesystem argument
+        "glob_directive",  # filesystem argument
+        "data_context_root_directory",  # filesystem argument
+        "bucket",  # s3 argument
+        "prefix",  # s3 argument
+        "delimiter",  # s3 argument
+        "max_keys",  # s3 argument
```

Comment:
Based on discussions with @Kilo59 we determined that we can avoid combinatorial enumeration of data assets for every storage system by sharing the settings between the `Datasource` and the add_<type>_asset() method, such that the `DataConnector` for the `DataAsset` could be instantiated in order to provide (via the top-level `DataConnector` interface methods) all utilities needed by `DataAsset` to access files on different storage environments.  As a consequence, the exclusions here in `Datasource` have to account for the settings of all supported storage environments.  In addition, the `Datasource.add_asset()` method here, in addition to the `asset` argument, must accept additional arguments (such as instantiated `DataConnector` and `test_connection_error_message`).

Reply:
Same response as above - https://github.com/great-expectations/great_expectations/pull/7197/files#r1114474090
These fields should not exist on the `DataAsset` they are simply use to create the appropriate `DataConnector`.

Reply:
@Kilo59 Please see above and let me know if you have guidance; otherwise, I keep running into "constructor does not know about this argument" type runtime errors.  Thank you.

Reply:
On further review, I understand why this is on the `Datasource`.
These are `_EXCLUDED_EXEC_ENG_ARGS`  on the `Datasource`
This is not the `_EXCLUDE_FROM_READER_OPTIONS` attribute on the pandas `DataAsset`.

Still, I think we could make this unnecessary by using a `data_connecter_options` / `data_connecter_parameter` object. I'm happy to save that for a followup PR though.